calicoctl/calico_ctl/diags.py

```diff
@@ -222,6 +121,27 @@ def save_diags(log_dir, runtime="docker"):
              (full_tar_path, os.path.basename(full_tar_path)))
 
 
+def write_diags(comment, command):
+    if comment:
+        print comment
+
+    try:
+        output = subprocess.check_output(command,
+                                         shell=True,
+                                         stderr=subprocess.STDOUT)
+        # Strip out non letters and numbers from the command to form the filename
+        filename = re.sub(r'[^a-zA-Z0-9 -]', "", command)
+
+        # And substitute underscore for spaces
+        filename = re.sub(r'\s', "_", filename)
+
+        with DiagsErrorWriter(temp_diags_dir, filename) as f:
```

Comment:
Using the DiagsFileWriter like this doesn't really buy you anything.  The point of that class was to protect against errors in the code writing the diags from trickling up the stack and causing the diags to bail out.  Better that the DiagsErrorWrite context manager encase as much of the processing as possible - especially the sub process call.